Simplify CMake thirdparty library target names

NewProggie opened the pull request:
In order to build platform independent static libraries, we can use CMakes
builtin `CMAKE_STATIC_LIBRARY_PREFIX` and `CMAKE_STATIC_LIBRARY_SUFFIX`
variables to separate the naming convention for Windows and Unix host
systems.

NewProggie committed:
Simplify CMake thirdparty library target names

In order to build platform independent static libraries, we can use CMakes
builtin `CMAKE_STATIC_LIBRARY_PREFIX` and `CMAKE_STATIC_LIBRARY_SUFFIX`
variables to separate the naming convention for Windows and Unix host
systems.

tensorflow-jenkins commented:
Can one of the admins verify this patch?

mrry reviewed (dismissed):
Thanks for cleaning this up!

mrry commented:
@tensorflow-jenkins test this please.

mrry commented:
@tensorflow-jenkins test this please.

mrry commented:
@tensorflow-jenkins test this please.

mrry commented:
@NewProggie It looks like `libjsoncpp.a` is the wrong name for the Linux CPU CMake build. Can you fix that up and we'll run the tests again? Thanks!

NewProggie committed:
Rollback jsoncpp CMake target to previous version

Since the jsoncpp thirdparty library is installed in a different location
for Windows, we cannot simplify the CMake configuration for this
dependency for now.

mrry commented:
@tensorflow-jenkins test this please.

mrry commented:
It looks like there's still a problem with the gRPC library name on Linux. My guess is that it only contains `Release/` on Windows.

NewProggie committed:
Rollback grpc CMake target to previous version

NewProggie commented:
Ah, shoot. I'll return to this one time in the future and fix all the cross references, as there's no technical reason for all the thirdparty libs to build in different subfolders (as far as I can see).

mrry commented:
OK - thanks for the effort here though. I think the inconsistency stems from the fact that some of the modules are built in tree (then we link the libraries directly from wherever the build output put them), whereas others are also installed (which puts them in a more predictable location).

It'd be great to sort this out at some point!

danmane commented:
Looks like this PR is dead at the moment, so I'm going to close it. Feel free to re-open or create a new one when its time to move forward.
